### ISSUE TITLE
Onnx model train using ORTTrainer

### ISSUE THREAD

**Issue by HemaSowjanyaMamidi** [microsoft/onnxruntime#9053]
Hi,
I know that ORTTrainer can be used to train pytorch models. Is there any way to train Onnx models using ORTTrainer?

**Comment by thiagocrepaldi** [comment]
Yes, you can load the onnx model and pass it to `ORTTrainer(model=onnx_obj)` constructor. You will also need to properly specify the `model_desc`. Checkout https://github.com/microsoft/onnxruntime/blob/master/orttraining/orttraining/python/training/orttrainer.py#L64-L120 for more details. https://github.com/microsoft/onnxruntime/blob/master/orttraining/orttraining/test/python/orttraining_test_orttrainer_bert_toy_onnx.py is a UT that loads an ONNX file and train it

If your ONNX `model` doesn't have an embedded `loss`, you will have to specify one as a `torch.nn.Module` instance

Hope that helps

**Comment by HemaSowjanyaMamidi** [comment]
Thankyou @thiagocrepaldi, for the resources. I tried training the model and encountered with an error stating loss function should not be specified when its an ONNX model. (I specified a torch.nn.functional code for loss function) 

Link:  https://github.com/microsoft/onnxruntime/blob/master/orttraining/orttraining/python/training/orttrainer.py#L153 
    elif isinstance(model, onnx.ModelProto):
                assert loss_fn is None, "'loss_fn' must not be specified when 'model' is an ONNX model" 

1. So how should I specify the loss function for an onnx model if I am creating the onnx model from scratch.
2. After training will the onnx model contains the optimizer weights which enables us to perform retraining or will the optimizer get discarded?(for example in keras we have the option include_optimizer=True/ False while saving the model).

**Comment by thiagocrepaldi** [comment]
@HemaSowjanyaMamidi 

1. yes, you are right. When a ONNX is specified as `model` for the `ORTTrainer` API, `model` must include loss nodes in the graph itself, connecting both forward and backward graphs. You have to edit your graph and add those nodes using https://github.com/onnx/onnx/blob/master/onnx/helper.py helpers to manipulate the graph, for instance

2. If you checkpoint your trained ONNX model, all params will stored in the ONNX file, including model, loss and optimizer's params. Checkout https://github.com/microsoft/onnxruntime/blob/master/orttraining/orttraining/python/training/orttrainer.py#L972-L1114 for info on the structure of the state and https://github.com/microsoft/onnxruntime/tree/master/orttraining/orttraining/test/python/checkpoint for examples on how to checkpoint
